src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala

```diff
@@ -0,0 +1,238 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.alignment
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.testing.UnitSpec
+
+class NeedlemanWunschAlignerTest extends UnitSpec {
+  /** Upper-cases and remove display-related characters from a string. */
+  def s(str: String): String = str.filterNot(ch => ch == '-').toUpperCase
+
+  /** Asserts a couple of things about an alignment that should be constant for global alignments. */
+  def assertValidGlobalAlignment(alignment: Alignment): Unit = {
+    alignment.queryStart  shouldBe 1
+    alignment.cigar.lengthOnQuery shouldBe alignment.query.length
+
+    alignment.targetStart shouldBe 1
+    alignment.cigar.lengthOnTarget shouldBe alignment.target.length
+  }
+
+  "NeedlemanWunschAligner.align" should "align two identical sequences with all matches" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1).align(s("ACGTAACC"), s("ACGTAACC"))
+    assertValidGlobalAlignment(result)
+    result.cigar.toString() shouldBe "8="
+    result.score shouldBe 8
+  }
+
+  it should "align two sequences with a single mismatch in them" in {
+    val q = s("AACCGGTT")
+    val t = s("AACCGtTT")
```

Comment:
Yes, they are currently.  The `s()` method in the test upper-cases everything and strips out the `-`s that I use to align the sequences for readability.  So the aligner actually sees: `AACCGGTT` and `AACCGTTT`, but I lower-case the T here so it's more visible where the mismatch is.